src/grpc/helium_unary_general_impl.erl

```diff
@@ -0,0 +1,205 @@
+config(
+    undefined = _Chain,
+    _Ctx,
+    #gateway_config_req_v1_pb{} = _Msg
+) ->
+    lager:debug("chain not ready, returning error response for msg ~p", [_Msg]),
+    {grpc_error, {grpcbox_stream:code_to_status(14), <<"temporarily unavailable">>}};
+config(
+    Chain,
+    Ctx,
+    #gateway_config_req_v1_pb{
+        keys = Keys
+    } = Request
+) ->
+    lager:debug("executing RPC config with msg ~p", [Request]),
+    Ledger = blockchain:ledger(Chain),
+    NumKeys = length(Keys),
+    Response0 =
+        case NumKeys > ?MAX_KEY_SIZE of
+            true ->
+                #gateway_error_resp_pb{
+                    error = <<"max_key_size_exceeded">>,
+                    details = list_to_binary(
+                        lists:concat(["limit ", ?MAX_KEY_SIZE, ". keys presented ", NumKeys])
+                    )
+                };
+            false ->
+                %% iterate over the keys submitted in the request and retrieve
+                %% current chain var value for each
+                Res =
+                    lists:map(
+                        fun(Key) ->
+                            try
+                                case
+                                    blockchain_ledger_v1:config(
+                                        list_to_existing_atom(Key),
+                                        Ledger
+                                    )
+                                of
+                                    {ok, V} -> to_var(Key, V);
+                                    {error, _} -> to_var(Key, undefined)
+                                end
+                            catch
+                                _:_ -> to_var(Key, undefined)
+                            end
+                        end,
+                        Keys
+                    ),
+                #gateway_config_resp_v1_pb{result = Res}
+        end,
+
+    Response1 = sibyl_utils:encode_gateway_resp_v1(
+        Response0,
+        sibyl_mgr:sigfun()
+    ),
+    {ok, Response1, Ctx}.
+
+-spec validators(
+    undefined | blockchain:blockchain(),
+    ctx:ctx(),
+    gateway_pb:gateway_validators_req_v1_pb()
+) -> {ok, gateway_pb:gateway_resp_v1_pb(), ctx:ctx()} | grpcbox_stream:grpc_error_response().
+validators(
+    undefined = _Chain,
+    _Ctx,
+    #gateway_validators_req_v1_pb{} = _Msg
+) ->
+    lager:debug("chain not ready, returning error response for msg ~p", [_Msg]),
+    {grpc_error, {grpcbox_stream:code_to_status(14), <<"temporarily unavailable">>}};
+validators(
+    _Chain,
+    Ctx,
+    #gateway_validators_req_v1_pb{
+        quantity = NumVals
+    } = Request
+) ->
+    lager:debug("executing RPC validators with msg ~p", [Request]),
+    %% get list of current validators from the cache
+    %% and then get a random sub set of these with size
+    %% equal to NumVals
+    IgnoreVals = application:get_env(sibyl, validator_ignore_list, []),
```

Comment:
We are going to end up needing a way to publish this information dynamically, right? We certainly don't want to necessarily do a software release for this

Reply:
Ignore vals here is not related to the deny list.  It is there to allow specified vals to be excluded from the list of validators returned to clients as part of the validators API.  This is useful if we know for example a validator which is running old code and we dont want it being used by clients.  Prob most useful on testnet where we have vals which just dont update.....no hard in leaving it in tho for time being